beacon_node/beacon_processor/src/lib.rs

```diff
@@ -762,205 +456,139 @@ impl<T: BeaconChainTypes> BeaconProcessorSend<T> {
                     &metrics::BEACON_PROCESSOR_SEND_ERROR_PER_WORK_TYPE,
                     &[work_type],
                 );
-                Err(Box::new(e))
+                Err(e)
             }
         }
     }
 }
 
-/// A consensus message (or multiple) from the network that requires processing.
-#[derive(Derivative)]
-#[derivative(Debug(bound = "T: BeaconChainTypes"))]
-pub enum Work<T: BeaconChainTypes> {
+pub type AsyncFn = Pin<Box<dyn Future<Output = ()> + Send + Sync>>;
+pub type BlockingFn = Box<dyn FnOnce() + Send + Sync>;
+pub type BlockingFnWithManualSendOnIdle = Box<dyn FnOnce(SendOnDrop) + Send + Sync>;
+
+/// Indicates the type of work to be performed and therefore its priority and
+/// queuing specifics.
+pub enum Work<E: EthSpec> {
     GossipAttestation {
-        message_id: MessageId,
-        peer_id: PeerId,
-        attestation: Box<Attestation<T::EthSpec>>,
-        subnet_id: SubnetId,
-        should_import: bool,
-        seen_timestamp: Duration,
+        attestation: GossipAttestationPackage<E>,
+        process_individual: Box<dyn FnOnce(GossipAttestationPackage<E>) + Send + Sync>,
+        process_batch: Box<dyn FnOnce(Vec<GossipAttestationPackage<E>>) + Send + Sync>,
```

Comment:
Objects which can be batch-verified (attestations and aggregated attestations) have two functions:

- `process_individual`: processes a single object.
- `process_batch`: processes multiple objects.

It is up to the `BeaconProcessor` to choose whether it should process objects individually or pull multiple objects from the queue and process them as a batch.